### DISCUSSION TITLE
List GitHub Contributors

### DISCUSSION THREAD

**Discussion by trueberryless** [withastro/starlight#975] [Feature Requests]
### What version of `starlight` are you using?

0.11.0

### What is your idea?

Create a component or some kind of integration for a list of contributors of the GitHub Repository linked under `astro.config.mjs`. This component can then be used for example on the start page of the website under "Next steps"...

### Why is this feature necessary?

If many people are working on a project, these people should all be appreciated on the website they are building. Therefore, a list with all contributors would make a lot of sense on websites built with Starlight.

### Do you have examples of this feature in other projects?

![image](https://github.com/withastro/starlight/assets/99918022/5d159543-2a11-4b8a-91e3-b148bb34855d)

### Participation

- [x] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Our component that does this in Astro’s docs could be a good reference:

- https://github.com/withastro/docs/blob/169708fd93a1d8595ef0c7decec645f8bd1a6b85/src/components/ContributorList.astro
- https://github.com/withastro/docs/blob/169708fd93a1d8595ef0c7decec645f8bd1a6b85/src/components/FacePile.astro
- https://github.com/withastro/docs/blob/169708fd93a1d8595ef0c7decec645f8bd1a6b85/src/util/getContributors.ts

**Reply by trueberryless** [reply]
How do you handle GitHub API tokens? Do I need a personal one?

**Reply by delucis** [reply]
Yes, you’d need to create one in your GitHub settings and then add it as an environment variable in your deployment provider (Netlify, Vercel, etc.) or locally if building on your machine.

**Comment by kevinzunigacuellar** [comment]
I think this will be a great built in component, +1!

**Reply by delucis** [reply]
I’m wondering if anyone’s built something generic like this anywhere? Like a component that works across GitHub, GitLab etc. It would kind of be a cool open-source project if it could be used in all kinds of contexts.

**Reply by HiDeoo** [reply]
> I think this will be a great built in component

_Built-in_ may add a lot of maintenance overhead if we were to support `git` (technically this is possible as we can get names & emails), GitHub, GitLab, etc. ?

**Reply by delucis** [reply]
Yeah, that’s partially why I was wondering about a standalone project that does this. Seems like a cool useful thing and isn’t necessarily tied to Starlight in terms of usefulness. A build step is probably important though, so maybe a good opportunity for an Astro component package for someone to build?

**Comment by trueberryless** [comment]
Looking at this discussion one year later, I have created a plugin already, which provides exactly this component: https://github.com/trueberryless-org/starlight-contributor-list
